checks.d/openstack.py

```diff
@@ -206,12 +244,34 @@ def get_hypervisor_stats(self):
 
         stats = {}
         for hyp in hypervisors:
-            stats[hyp] = {
-                'payload': self.get_stats_for_single_hypervisor(hyp),
-                'uptime': self.get_uptime_for_single_hypervisor(hyp)
-            }
+            stats[hyp] = {}
+            try:
+                stats[hyp]['payload'] = self.get_stats_for_single_hypervisor(hyp)
+            except:
+                self.warning('Unable to get stats for hypervisor {0}.'.format(hyp))
+
+            try:
+                stats[hyp]['uptime'] = self.get_uptime_for_single_hypervisor(hyp)
+            except:
```

Comment:
Do we know enough about the failure modes of these functions to catch named `Exception`s here? My guess is we don't, in which case maybe it's prudent to log the stringified `Exception` in the warning message so that we know what went wrong.

Ditto for the other bare `except`s in this file.